.github/workflows/cmake_conan.yml

```diff
@@ -11,6 +11,11 @@ jobs:
           - os: ubuntu-latest
     steps:
       - uses: actions/checkout@v3.5.2
+      - uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: amd64
+        if: ${{ matrix.os == 'windows-2019' }}
+        name: setup msvc vcvars
```

Comment:
I am not so sure about this. Does this automatically setup the VS prompt with the compiler activated? If that is the case, the tests should run without it, it shouldn't be necessary to run in an activated prompt.

Reply:
cmake with the ninja generator (or any other generator that are not the visual studio) require to run in an activated prompt for compiler detection (so that it detects cl.exe). 

If the goal is be able to invoke pytest in a non-activated prompt - this would complicate the test file further than necessary? - the environment needs to be activated upon invocation of the cmake process, and in order to reliable locate the VS installation on a generic machine we'd have to use vswhere and the likes (more logic). 

perhaps we could use Conan to generate the activation script (since Conan has the logic to deal with locating the visual studio installation, and either capture the environment in a session-scope fixture (in the same way we are doing for `tmpdirs`), or if we have located the activation script, modify the `run` method such that it loads it first ahead of running the command (I think it should work since we have `shell=True`, but I'm not sure _which_ shell python uses in this instance on Windows and if that works reliably with powershell/cmd/bash).